pkg/tfvars/openstack/rhcos_image.go

```diff
@@ -1,175 +0,0 @@
-package openstack
-
-import (
-	"fmt"
-	"net/url"
-	"os"
-	"path/filepath"
-	"time"
-
-	"github.com/gophercloud/gophercloud"
-	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
-	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imageimport"
-	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
-	"github.com/sirupsen/logrus"
-
-	"github.com/openshift/installer/pkg/rhcos/cache"
-	openstackdefaults "github.com/openshift/installer/pkg/types/openstack/defaults"
-)
-
-// uploadBaseImage creates a new image in Glance and uploads the RHCOS image there
-func uploadBaseImage(cloud string, baseImage, imageName string, infraID string, imageProperties map[string]string) error {
-	var localFilePath string
-
-	url, err := url.Parse(baseImage)
-	if err != nil {
-		return err
-	}
-
-	// We support 'http(s)' and 'file' schemes. If the scheme is http(s), then we will upload a file from that
-	// location. Otherwise will take local file path from the URL.
-	switch url.Scheme {
-	case "http", "https":
-		localFilePath, err = cache.DownloadImageFile(baseImage, cache.InstallerApplicationName)
-		if err != nil {
-			return err
-		}
-	case "file":
-		localFilePath = filepath.FromSlash(url.Path)
-	default:
-		return fmt.Errorf("unsupported URL scheme: %q", url.Scheme)
-	}
-
-	logrus.Debugln("Creating a Glance image for RHCOS...")
-
-	f, err := os.Open(localFilePath)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	conn, err := openstackdefaults.NewServiceClient("image", openstackdefaults.DefaultClientOpts(cloud))
-	if err != nil {
-		return err
-	}
-
-	// By default we use "qcow2" disk format, but if the file extension is "raw",
-	// then we set the disk format as "raw".
-	diskFormat := "qcow2"
-	if extension := filepath.Ext(localFilePath); extension == "raw" {
-		diskFormat = "raw"
-	}
-
-	imageCreateOpts := images.CreateOpts{
-		Name:            imageName,
-		ContainerFormat: "bare",
-		DiskFormat:      diskFormat,
-		Tags:            []string{fmt.Sprintf("openshiftClusterID=%s", infraID)},
-		Properties:      imageProperties,
-		// TODO(mfedosin): add Description when gophercloud supports it.
-	}
-
-	img, err := images.Create(conn, imageCreateOpts).Extract()
-	if err != nil {
-		return err
-	}
-	logrus.Debugf("Image %s was created.", img.Name)
-
-	// FIXME(mfedosin): We have to temporary disable image import, because it looks
-	// like there are problems on the server side.
-	// Revert this patch when the problems are fixed.
-	// https://github.com/openshift/installer/issues/3403
-	// useImageImport, err := isImageImportSupported(cloud)
-	// if err != nil {
-	// 	return err
-	// }
-	useImageImport := false
-
-	if useImageImport {
```

Comment:
Should the image import support really be removed even if it defaults to false?

Reply:
It's dead code. Why should it be in?